orchestrator/test_runner/src/upgrade.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{utils::*, happy_path_test, happy_path_test_v2};
+use clarity::Address as EthAddress;
+use deep_space::Contact;
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use deep_space::client::ChainStatus;
+use tonic::transport::Channel;
+use web30::client::Web3;
+use gravity_proto::cosmos_sdk_proto::cosmos::bank::v1beta1::Metadata;
+use crate::ibc_metadata::submit_and_pass_ibc_metadata_proposal;
+
+/// Perform a series of integration tests to seed the system with data, then submit and pass a chain
+/// upgrade proposal
```

Comment:
I see the strategy here, very clever way to shortcut a lot of the upgrade test complexity, hopefully we can move this to more rust-managed.

Hopefully the plan outlined here actually works well https://github.com/Gravity-Bridge/Gravity-Bridge/issues/129